compiler/src/dotty/tools/dotc/transform/NonLocalReturns.scala

```diff
@@ -95,10 +95,11 @@ class NonLocalReturns extends MiniPhase {
 
   override def transformReturn(tree: Return)(using Context): Tree =
     if isNonLocalReturn(tree) then
-      report.errorOrMigrationWarning(
+      report.gradualErrorOrMigrationWarning(
           "Non local returns are no longer supported; use scala.util.control.NonLocalReturns instead",
           tree.srcPos,
-          from = future)
+          warnFrom = `3.0`,
```

Comment:
```suggestion
          warnFrom = `3.2`,
```

Reply:
It used to warn if you weren't in a "Scala 2" mode or frame of mind. That is `3.0-migration` now? But I see that this warns under `3.0-migration`. I expected `X-migration` semantics: "error or migration warning", and "warning or free pass if migrating at X".

Reply:
Pushed that conjecture.